### PR TITLE
fix: zod error message and missing translation

### LINKED ISSUE #14785: [CAL-3570] missing translation: /signup
I consent ...

By proceeding, you agree ...

image

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3570](https://linear.app/calcom/issue/CAL-3570/missing-translation-signup)</sub>

### LINKED ISSUE #14786: [CAL-3571] bad error for SAML if email is empty
image

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3571](https://linear.app/calcom/issue/CAL-3571/bad-error-for-saml-if-email-is-empty)</sub>

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

Fixes #14785 
Fixes #14786 
/claim #14877

### CODE REVIEW

File: packages/features/bookings/Booker/components/BookEventForm/BookEventForm.tsx [hunk 1]
```diff
@@ -115,13 +115,13 @@ export const BookEventForm = ({
         {!isPlatform && IS_CALCOM && (
           <div className="text-subtle my-3 w-full text-xs opacity-80">
             <Trans i18nKey="signing_up_terms">
-              By proceeding, you agree to our{" "}
+              {t("agree_to_our")}{" "}
```
Comment (Udit-takkar) [inline packages/features/bookings/Booker/components/BookEventForm/BookEventForm.tsx:118]: This is not how you should use i18n here. check Trans https://react.i18next.com/latest/trans-component

File: apps/web/lib/signup/getServerSideProps.tsx [hunk 2]
```diff
@@ -17,9 +17,10 @@ const checkValidEmail = (email: string) => z.string().email().safeParse(email).s
 const querySchema = z.object({
   username: z
     .string()
+    .min(1)
     .optional()
     .transform((val) => val || ""),
-  email: z.string().email().optional(),
+  email: z.string().email().min(1).optional(),
```
Comment (Udit-takkar) [inline apps/web/lib/signup/getServerSideProps.tsx:23]: no we don't want to add min(1) here. the issue is about improving the error message

Comment (Udit-takkar) [reply apps/web/lib/signup/getServerSideProps.tsx:23]: something like "Invalid Email"

Comment (anikdhabal) [reply apps/web/lib/signup/getServerSideProps.tsx:23]: Got it. Yeah, I improved the messages in the schema.

**Review by Udit-takkar — APPROVED** [review]

**Comment by anikdhabal** [comment]
Screenshot 2024-05-03 225725

@Udit-takkar the problem is not solved. I think my changes were valid.

**Review by PeerRich — APPROVED** [review]

**Pull request merged**